cmd/kube-proxy/app/server.go

```diff
@@ -86,6 +88,10 @@ import (
 	"k8s.io/utils/pointer"
 )
 
+func init() {
+	utilruntime.Must(metricsfeatures.AddFeatureGates(utilfeature.DefaultMutableFeatureGate))
```

Comment:
why is this needed?

Reply:
You need to register the feature gates before flags are applied, so conventionally this is done through an init function.

Reply:
are we not setting them on line 257?

```
	return utilfeature.DefaultMutableFeatureGate.SetFromMap(o.config.FeatureGates)
```

Reply:
We can probably also try setting it the line before. But the init function is pretty standard way of setting this across the kubernetes code base.

Reply:
I was able to curl the /metrics/slis endpoint after commenting out the init() block- so functionally, its ok to be removed, but it makes sense to keep the init() block and follow the conventional way for initialization purposes.

Reply:
Your call @aojea 😃 we're open to either way, but I think we're slightly more inclined to follow convention.

Reply:
yeah, me too, I just wanted to fully understand it
Thanks for clarifying